x-pack/plugins/file_upload/public/components/json_upload_and_parse.tsx

```diff
@@ -74,6 +77,26 @@ export class JsonUploadAndParse extends Component<FileUploadComponentProps, Stat
       return;
     }
 
+    //
+    // check permissions
+    //
+    const canImport = await hasImportPermission({
```

Comment:
Should we short-circuit this if `this.state.indexName` isn't defined? Looks like it will still do the endpoint roundtrip even if it's an empty string

Reply:
The button triggering import is only enabled when indexName is set so no need to check for empty string. Logic ensuring this is in _onIndexNameChange